### PR TITLE
Add support to test a custom state test

### PR DESCRIPTION
**Pull request by jwasinger**
This PR adds the ability to test against custom-generated state tests.  The argument `--customStateTest` is used to specify the relative path of a file containing a custom test.

This PR is necessary for implementing trace equivalence testing and fuzz testing with EthereumjsVM.

Depends on changes in https://github.com/ethereumjs/ethereumjs-testing/tree/test-from-source

### CODE REVIEW

File: tests/GeneralStateTestsRunner.js [hunk 1]
```diff
@@ -72,6 +72,7 @@ function runTestCase (testData, t, cb) {
 }
 
 module.exports = function runStateTest (options, testData, t, cb) {
+  debugger
```
Comment (axic) [inline tests/GeneralStateTestsRunner.js]: Is this some new ES* magic?

Comment (julianduque) [reply tests/GeneralStateTestsRunner.js]: since `testState` is being run with `node inspect` this acts as a breakpoint for the debugger.

**Comment by holgerd77** [comment]
I've now looked 2-3 times over this PR and I still have difficulties to get the essence of it respectively grab what the PR wants to achieve. Can you come up with a practical example? Is this something different that can be achieved with either the new ``testsPath`` flag (put your custom test in a separate folder) or the existing ``getSingleFile`` method in ``ethereumjs-testing``?

**Comment by jwasinger** [comment]
Yes.  This is necessary for fuzz testing.  Even with the current CLI options, in order to run a state test, it has to be located under the tests file tree.

**Comment by holgerd77** [comment]
Sorry, I'm not really getting it, can' you just make a directory ``../MyFuzzTests/tests/fuzztest1.json`` and the use the new ``--testsPath`` option with ``../MyFuzzTests/`` - and eventually with ``--test='fuzztest1'``?

**Comment by holgerd77** [comment]
Can you give a short comment on my last question?

**Comment by jwasinger** [comment]
I changed the command line argument name to 'customStateTest'.  We could potentially do what you are suggesting.  But IMO it would be easier to have a separate command line option to directly pass the test to tester.js.

The alternative would be to modify this code to do what you are suggesting:
https://github.com/jwasinger/evmlab/blob/b3f8f48591de6320bba78230290bcf24968bfc02/trace_statetests.py#L358-L369

I'll see if I can do some modifications to get it working without the new command line argument.

**Pull request merged**